src/woo_publications/publications/formset.py

```diff
@@ -53,3 +54,10 @@ def delete_existing(self, obj, commit=True):
                 partial(remove_document_from_index.delay, document_id=obj.pk)
             )
         super().delete_existing(obj, commit)
+
+
+class ThroughModelInlineFormset(BaseInlineFormSet):
+    def __init__(self, *args, **kwargs):
+        assert "_django_user" in kwargs
+        del kwargs["_django_user"]
```

Comment:
why is this necessary?

Reply:
This is necessary because of the `AdminAuditLogMixin` adds `_django_user` to the formset in the `get_formset_kwargs`

https://github.com/GPP-Woo/GPP-publicatiebank/blob/main/src/woo_publications/logging/admin_tools.py#L98

Reply:
the proper fix is to check the `inline` argument in the linked code and only apply the extra kwarg to those inline classes that expect it - we can follow the relationships to make this loosely coupled:

```py
...
requires_user = issubclass(inline.formset, AuditLogInlineformset)
if requires_user:
    kwargs["_django_user"] = request.user
...
```

`inline` in this case is the instance of the admin inline (`DocumentInlineAdmin`), for which we can check the `formset` class it uses (which we set to a subclass of `AuditLogInlineformset` as that one expects the user kwarg).

I had some version of that initially but couldn't get any test coverage for the branches since we only had one inline in a model admin with audit logging :) but that has now changed.